homeassistant/components/tessie/__init__.py

```diff
@@ -94,41 +95,48 @@ async def async_setup_entry(hass: HomeAssistant, entry: TessieConfigEntry) -> bo
 
     # Energy Sites
     tessie = Tessie(session, api_key)
+    energysites: list[TessieEnergyData] = []
+
     try:
-        products = (await tessie.products())["response"]
+        scopes = await tessie.scopes()
     except TeslaFleetError as e:
         raise ConfigEntryNotReady from e
 
-    energysites: list[TessieEnergyData] = []
-    for product in products:
-        if "energy_site_id" in product:
-            site_id = product["energy_site_id"]
-            api = EnergySpecific(tessie.energy, site_id)
-            energysites.append(
-                TessieEnergyData(
-                    api=api,
-                    id=site_id,
-                    live_coordinator=TessieEnergySiteLiveCoordinator(hass, api),
-                    info_coordinator=TessieEnergySiteInfoCoordinator(hass, api),
-                    device=DeviceInfo(
-                        identifiers={(DOMAIN, str(site_id))},
-                        manufacturer="Tesla",
-                        name=product.get("site_name", "Energy Site"),
-                    ),
+    if Scope.ENERGY_DEVICE_DATA in scopes:
+        try:
+            products = (await tessie.products())["response"]
+        except TeslaFleetError as e:
+            raise ConfigEntryNotReady from e
```

Comment:
What are the possible errors that can return? Is it justified that the ConfigEnteyNotReady is used here?

Reply:
The possible errors:

https://github.com/Teslemetry/python-tesla-fleet-api/blob/main/tesla_fleet_api/exceptions.py

Is Config Entry Not Ready justified, I believe so, otherwise the entities won't work at all.

Also this code block isn't not new in this PR and unrelated to the issue I'm fixing.

Reply:
If the scope isn't set correctly, should it trigger a re-auth instead?

Reply:
Thanks for clarifying!